File: frontend/src/api/credits.ts
import { apiRequest } from "./client";

export interface Credit {
  person_id: string; name: string; image: string | null; person_url: string | null;
  character_id: string | null; character: string | null; character_image: string | null;
  category: "cast" | "crew"; role: string; language: string | null;
  dub_group: string | null; notes: string | null; order: number;
}
export interface CreditSource {
  provider: string; external_id: string; title: string; source_url: string;
  original_language: string | null; fetched_at: string | null; credits: Credit[];
}
export interface SeriesCredits { catalog_id: string | null; original_language: string | null; sources: CreditSource[]; }
export interface CreditMatch { id: string; title: string; year: number | null; image: string | null; }
export interface CreditSettings { tmdb_configured: boolean; tvdb_configured: boolean; }
const path = (server: number, item: string) => `/servers/${server}/items/${encodeURIComponent(item)}/credits`;
export const creditsApi = {
  get: (server: number, item: string) => apiRequest<SeriesCredits>(path(server, item)),
  import: (server: number, item: string, provider: string, external_id: string) => apiRequest<SeriesCredits>(path(server, item), { method: "POST", body: JSON.stringify({ provider, external_id }) }),
  language: (server: number, item: string, original_language: string | null) => apiRequest<SeriesCredits>(path(server, item), { method: "PUT", body: JSON.stringify({ original_language }) }),
  remove: (server: number, item: string, provider: string, external_id: string) => apiRequest<SeriesCredits>(`${path(server, item)}?${new URLSearchParams({ provider, external_id })}`, { method: "DELETE" }),
  search: (provider: string, query: string) => apiRequest<CreditMatch[]>(`/credits/search?${new URLSearchParams({ provider, query })}`),
  settings: () => apiRequest<CreditSettings>("/credits/settings"),
  saveToken: (tmdb_access_token: string) => apiRequest<CreditSettings>("/credits/settings", { method: "PUT", body: JSON.stringify({ tmdb_access_token }) }),
};

File: crates/posterview-contracts/src/credits.rs
use serde::{Deserialize, Serialize};

/// A provider-scoped credit. Language describes the performance, never UI text.
#[derive(Debug, Clone, Default, PartialEq, Eq, Serialize, Deserialize)]
pub struct Credit {
    pub person_id: String,
    pub name: String,
    pub image: Option<String>,
    pub person_url: Option<String>,
    pub character_id: Option<String>,
    pub character: Option<String>,
    pub character_image: Option<String>,
    pub category: String,
    pub role: String,
    pub language: Option<String>,
    pub dub_group: Option<String>,
    pub notes: Option<String>,
    pub order: i64,
}

#[derive(Debug, Clone, Default, PartialEq, Eq, Serialize, Deserialize)]
pub struct CreditSource {
    pub provider: String,
    pub external_id: String,
    pub title: String,
    pub source_url: String,
    pub original_language: Option<String>,
    pub fetched_at: Option<String>,
    pub credits: Vec<Credit>,
}

#[derive(Debug, Clone, Default, Serialize, Deserialize)]
pub struct SeriesCredits {
    pub catalog_id: Option<String>,
    pub original_language: Option<String>,
    pub sources: Vec<CreditSource>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreditSearchResult {
    pub id: String,
    pub title: String,
    pub year: Option<i64>,
    pub image: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreditProviderSettings {
    pub tmdb_configured: bool,
    pub tvdb_configured: bool,
}
